### ISSUE TITLE
Re-implement the --regroup option for Aegean to be faster and stand-alone

### ISSUE THREAD

**Issue by PaulHancock** [PaulHancock/Aegean#150]
The Aegean priorized fitting option `--regroup` serves a vital purpose but is so slow that I always recommend that people don't use it.

The purpose is to ensure that sources that are near to each other are fit simultaneously. This is particularly useful when you have a reference catalogue that comes from some high resolution image, and you want to do priorized fitting on a lower resolution image.

The regrouping is slow because I home-brewed a solution after not being able to find some ready built thing. Now however there seems to be some drop-in replacements for what I want to do in the form of `sklearn.neighbors.BallTree` and the `scipy.spatial.distance.haversine` metric

**Comment by PaulHancock** [comment]
Branch [issue150](/PaulHancock/Aegean/tree/PaulHancock/issue150) is being used to track this issue.

**Comment by PaulHancock** [comment]
Fixed in PR #152
